scipy/stats/_binned_statistic.py

```diff
@@ -68,24 +68,41 @@ def binned_statistic(x, values, statistic='mean',
     Notes
     -----
     All but the last (righthand-most) bin is half-open.  In other words, if
-    `bins` is::
-
-      [1, 2, 3, 4]
-
-    then the first bin is ``[1, 2)`` (including 1, but excluding 2) and the
-    second ``[2, 3)``.  The last bin, however, is ``[3, 4]``, which *includes*
-    4.
+    `bins` is ``[1, 2, 3, 4]``, then the first bin is ``[1, 2)`` (including 1,
+    but excluding 2) and the second ``[2, 3)``.  The last bin, however, is
+    ``[3, 4]``, which *includes* 4.
 
     .. versionadded:: 0.11.0
 
     Examples
     --------
+    >>> from scipy import stats
+    >>> import matplotlib.pyplot as plt
+
+    First a basic example:
+
     >>> stats.binned_statistic([1, 2, 1, 2, 4], np.arange(5), statistic='mean',
-    ... bins=3)
+    ...                        bins=3)
     (array([ 1.,  2.,  4.]), array([ 1.,  2.,  3.,  4.]), array([1, 2, 1, 2, 3]))
 
-    >>> stats.binned_statistic([1, 2, 1, 2, 4], np.arange(5), statistic='mean', bins=3)
-    (array([ 1.,  2.,  4.]), array([ 1.,  2.,  3.,  4.]), array([1, 2, 1, 2, 3]))
+    Now use ``bin_edges`` and ``binnumber`` to make a plot of a distribution
+    that shows the mean and distribution around that mean per bin, on top of a
+    regular histogram and the probability distribution function:
+
+    >>> x = np.linspace(0, 5, num=500)
+    >>> x_pdf = stats.maxwell.pdf(x)
+    >>> samples = stats.maxwell.rvs(size=10000)
+
+    >>> bin_means, bin_edges, binnumber = stats.binned_statistic(x, x_pdf,
+    ...         statistic='mean', bins=25)
```

Comment:
Would it not make more sense to calculate the mean of the `samples`? The example works this way, too, but it's not too common to calculate the mean of the points on the pdf, since the pdf is known analytically and I can evaluate it at any point I want anyway. On the other hand, the `samples` is what I would measure in an experiment and then I would want to compare the mean of the samples in each bin with the red line (the analytical pdf).

Just for the record: I think we are all well aware that in many practical cases, it is better not use a binned statistic to check if a measured distribution matches e.g. a maxwell distribution, but I this is an easy to grasp example that greatly helps to understand what the return values of this function are.